lib/facility.rb

```diff
@@ -1,14 +1,73 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
+    @services
   end
+
+  def collection_fee_assignment
+    {
+    :ev => 200,
+    :antique => 25,
+    :regular => 100
+    }
+  end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      @registered_vehicles << vehicle
+      vehicle.update_registration_date
+      @collected_fees += collection_fee_assignment[vehicle.plate_type]
+      @registered_vehicles
+    else 
+      nil
```

Comment:
This is not explicitly necessary. If the `if` condition is not met, by default this method will do nothing and return `nil`.